### PR TITLE
Allow getting/setting publicPreview flag through API for Emails

### PR DESCRIPTION
**Pull request by galvani**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [X]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR allow to get and set email flag publicPreview through API.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send API request for an email: 
```
GET {{host}}/api/emails/{email_id}
Accept: application/json
Authorization: Bearer {{auth_token}}
```
the response will containt
```
"publicPreview": true,
```

update existing email by putting 
```
"publicPreview": true,
```
into payload body

the post value can be either:
true: true,"T","Y"
false: false,"F","N"

and verify it has been updated in UI

### CODE REVIEW

File: app/bundles/EmailBundle/Controller/Api/EmailApiController.php [hunk 1]
```diff
@@ -214,4 +215,13 @@ public function replyAction(Reply $replyService, RandomHelperInterface $randomHe
             $this->view(['success' => true], Response::HTTP_CREATED)
         );
     }
+
+    protected function prepareParametersFromRequest(FormInterface $form, array &$params, object $entity = null, array $masks = [], array $fields = []): void
+    {
+        if (isset($params['publicPreview']) && $entity instanceof Email) {
+            $entity->setPublicPreview(InputHelper::boolean($params['publicPreview']) ?? false);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Controller/Api/EmailApiController.php]: Didn't you mean this instead?
```suggestion
            $entity->setPublicPreview(InputHelper::boolean($params['publicPreview'] ?? false));
```

Comment (kuzmany) [reply app/bundles/EmailBundle/Controller/Api/EmailApiController.php]: @escopecz good catch

Comment (kuzmany) [reply app/bundles/EmailBundle/Controller/Api/EmailApiController.php]: I am gonna revert the commit, Based on code, isset($params['publicPreview']) is in parent condition and this InputHelper::booleen in some cases can return null. Then it's corret, always set bool in setPublicPreview

**Comment by kuzmany** [comment]
Due wrong force push I had to push new PR https://github.com/mautic/mautic/pull/13609

**Pull request closed without merging**